src/prefect/client/client.py

```diff
@@ -567,6 +567,34 @@ def set_flow_run_state(
 
         self.graphql(mutation, state=serialized_state)  # type: Any
 
+    def get_latest_cached_states(
+        self, task_id: str, created_after: datetime.datetime
+    ) -> List["prefect.engine.state.State"]:
+        """
+        Pulls all Cached states for the given task which were created after the provided date.
+
+        Args:
+            - task_id (str): the task id for this task run
+            - created_after (datetime.datetime): the earliest date the state should have been created at
+
+        Returns:
+            - List[State]: a list of Cached states created after the given date
+        """
+        where_clause = {
+            "where": {
+                "state": {"_eq": "Cached"},
+                "task_id": {"_eq": task_id},
+                "created": {"_gte": created_after.isoformat()},
+            }
+        }
+        query = {"query": {with_args("task_run", where_clause): "serialized_state"}}
```

Comment:
I think this needs an `order_by` clause; later on these results are iterated over and the first match is returned, but without sorting that might not be the most current cache (although on second thought it's hard to imagine how there could be two valid caches 🤔 )

Reply:
Yea, so -- I originally intended to limit + order_by, but `with_args` only accepts a single argument type, and I determined if multiple caches are valid, it shouldn't matter which is returned (also, I don't think that's possible).

Reply:
I think you're right, shouldn't make a practical difference.

For future reference, just pass a dictionary to `with_args`:

```python
with_args(task_run, {'where': {'x': {'_eq': 1}}, 'order_by': 2, 'limit': 3})
```

results in
```graphql
task_run(where: {x: {_eq: 1}}, order_by: 2, limit: 3)
```

Reply:
Ah nice, updated!